geometry/triangulate.cpp

```diff
@@ -169,19 +171,52 @@ bool pointInMultiPolygon(const math::Point2d &test,
     return c;
 }
 
-} // namespace
+// Converts input coordinates to interval [0, 2^16] and vice versa
+// (work around boost polygon integer nature)
+class CoordConverter {
+    math::Point2 offset_;
+    double mult_;
+    double invMult_;
 
-namespace bp = boost::polygon;
+public:
+    explicit CoordConverter(const math::MultiPolygon& mpoly)
+    {
+        math::Extents2 ext(math::InvalidExtents{});
+        for (const auto& poly : mpoly) {
+            for (const auto& p : poly) {
+                math::update(ext, p);
+            }
+        }
+        auto size = math::size(ext);
+        const double MaxValue = 1 << 16;
+
+        mult_ = MaxValue / std::max(size.width, size.height);
+        invMult_ = 1.0 / mult_;
+        offset_ = ext.ll;
+    }
 
-const double Mult = 1 << 16; // work around boost polygon integer nature
-const double InvMult = 1.0 / Mult;
+    inline bp::point_data<double> operator()(const math::Point2& p) const
+    {
+        return bp::point_data<double>((p(0) - offset_(0)) * mult_,
+                                      (p(1) - offset_(1)) * mult_);
+    }
+
+    inline math::Point2 operator()(const bp::point_data<double>& p) const
+    {
+        return math::Point2(p.x() * invMult_ + offset_(0),
+                            p.y() * invMult_ + offset_(1));
+    }
```

Comment:
@pavelsevecek @vaclavblazek the forward backward conversion depends only on type, which look cryptic below in the code. For me, `cnv(p)` and `cnv.inv(v)` would be more descriptive.

Reply:
will merge as is, possibly fix in next MR